### PR TITLE
ci: Build image from Containerfile when tag changes

### PR DESCRIPTION
**Pull request by swick**
This is based on work by @flexagoon (https://github.com/flatpak/xdg-desktop-portal/pull/1270) but changes how the build images are being build, has a few things renamed, fixes an issue with the artifact upload, and adds a check for POTFILES.in.

### CODE REVIEW

**Comment by swick** [comment]
I think someone needs to create and adjust the permissions on the xdg-desktop-portal-build container package in the flatpak org.

**Comment by swick** [comment]
Rebased. Can I convince anyone to take a look at this?

**Comment by GeorgesStavracas** [comment]
Can I ask you to do some test releases on your repo, so we can see how it goes?

**Comment by swick** [comment]
There was a whole bunch of broken things and I spent most of the day on fixing and testing them. It now uses a single container for testing, docs and releases. Releases work (https://github.com/swick/xdg-desktop-portal/actions/runs/11239635794).

File: tests/asan.suppression [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# external (GIO?)
+leak:g_dbus_message_new_from_blob
+# Bugs in our code
+# Take a look at them and try to figure out  what's going on!
```
Comment (hfiguiere) [inline tests/asan.suppression:4]: some are in libportal. For example https://github.com/flatpak/libportal/pull/170/files

File: .github/workflows/container.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ github.token }}
+
+      - name: Check if image already exists on GHCR
+        id: check
+        run: |
+          image=ghcr.io/${{ github.repository_owner }}/xdg-desktop-portal-build:${{ env.IMAGE_TAG }}
```
Comment (GeorgesStavracas) [inline .github/workflows/container.yml:35]:
```suggestion
          image=ghcr.io/${{ github.repository_owner }}/xdg-desktop-portal:${{ env.IMAGE_TAG }}
```

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+      - name: Tag release
+        if: ${{ !fromJSON(github.event.inputs.dryRun) }}
+        run: |
+          git tag $releaseVersion
+          git push --tags
+
+      - name: Create release
+        if: ${{ !fromJSON(github.event.inputs.dryRun) }}
+        uses: ncipollo/release-action@v1.13.0
```
Comment (GeorgesStavracas) [inline .github/workflows/release.yml:96]:
```suggestion
        uses: ncipollo/release-action@v1.14.0
```

**Review by GeorgesStavracas — CHANGES_REQUESTED** [review]
Small comments, LGTM otherwise

**Comment by swick** [comment]
@GeorgesStavracas do you mind running the workflows to see if everything works?

**Comment by swick** [comment]
As requested on matrix, this is now without the release workflow and POTFILES checks.

**Review by GeorgesStavracas — APPROVED** [review]

**Pull request merged**